### ISSUE TITLE
module not found when running a fresh install of remix 1.15

### ISSUE THREAD

**Issue by michaelgainyo** [remix-run/remix#6078]
### What version of Remix are you using?

1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. run `npx create-remix@latest`
2. pick just the basics, express server, typescript following the prompt and install
3. `npm run dev` and it automatically crashes with the following:

```
Building Remix app in production mode...
Built in 178ms
[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): server.js   // <--- HERE
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `cross-env NODE_ENV=production node ./server.js --require dotenv/config ./server.js` // <--- HERE
node:internal/modules/cjs/loader:1056
  throw err;
  ^

Error: Cannot find module '/path/to/project/test/server.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1053:15)
    at Module._load (node:internal/modules/cjs/loader:898:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:84:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v19.4.0
[nodemon] app crashed - waiting for file changes before starting...
Watching Remix app in development mode...
💿 Built in 180ms
```

### Expected Behavior

Should run

### Actual Behavior

I'm guessing the issue lies with the `start` and `dev:node` scripts where nodemon is watching for a non-existing `server.js` file when in fact, it's `server.ts`

```
├── test
│   ├── README.md
│   ├── app
│   │   ├── entry.client.tsx
│   │   ├── entry.server.tsx
│   │   ├── root.tsx
│   │   └── routes
│   │       └── _index.tsx
│   ├── build
│   │   ├── index.js
│   │   └── index.js.map
│   ├── package.json
│   ├── public
│   │   ├── build
│   │   └── favicon.ico
│   ├── remix.config.js
│   ├── remix.env.d.ts
│   ├── server.ts         // <- HERE
│   └── tsconfig.json
```

**Comment by tom-dudman** [comment]
I got round this by installing `ts-node` and changing the `dev:node` script in `package.json` to:

`"dev:node": "cross-env NODE_ENV=development nodemon --require dotenv/config ./server.ts --watch ./server.ts"`

I don't know if this an optimal fix but it unblocked me for now.

**Comment by oldfeel** [comment]
The 'start' command needs to be changed like this
`"start": "cross-env NODE_ENV=production ts-node ./server.ts"`

**Comment by michaelgainyo** [comment]
@oldfeel @tom-dudman yes installing ts-node does the trick

**Comment by tom-dudman** [comment]
> The problem seems to be that `server.js` never gets built?

I'd expect `server.js` to be built by running `npm run build` so that `npm start` would work (as per `README.md`)

I don't know about `npm run dev` - would you expect `index.js` to be output from running that?

**Comment by Download** [comment]
> I'd expect `server.js` to be built by running `npm run build`

Yes. I cannot imagine that we need to use `node-ts` in production...

> I don't know about npm run dev - would you expect index.js to be output from running that?

I am not sure. I have not been using Remix for long enough. Other tools often don't output files at all in `dev` mode and instead just keep them in memory... I did see some other issues talking about how they had to run `npm run build` at least once before using `dev` mode.... 

I just tested `create-remix` choosing the Remix App Server i.s.o. Express and then it does work. It does not generate a `server.js` though. But I am guessing that's because the Express version exposes that `server.ts`, whereas Remix App Server hides it behind the curtains.

Also, I think some maintainer should remove the `bug:unverified` label because we already have 6 verifications of it and it is very easy to reproduce it by just following the Getting Started guide and picking `Express` as the server.
